test/functional/test_rate_limits.py

```diff
@@ -2,9 +2,12 @@
 
 from __future__ import unicode_literals
 
+from mock import patch
+
 
 def test_too_many_requests_causes_retry(box_client, mock_box, monkeypatch):
     monkeypatch.setattr(mock_box, 'RATE_LIMIT_THRESHOLD', 1)
-    box_client.folder('0').get()
-    box_client.folder('0').get()
+    with patch('random.uniform', return_value=1):
```

Comment:
You should also leave a comment here on why it's important to have the randomness stubbed out to the highest possible value (to avoid the race condition)